routes/web-admin.php

```diff
@@ -20,6 +22,15 @@
         Route::post('/',[LoginController::class, 'login'])->name('login');
     });
     
+    Route::get('/tasks', [TaskController::class, 'showAll'])->name('tasks');
+    Route::get('/users', [UserManagementController::class, 'showAll'])->name('users');
+    Route::get('/add/user', [UserManagementController::class, 'showAdduser'])->name('add');
+    Route::post('/add/user', [UserManagementController::class, 'insert'])->name('adduser');
+    Route::get('/show/{user}', [UserManagementController::class, 'showUser'] )->name('showuser');
+    Route::get('/tasks/{user}', [UserManagementController::class, 'showTasks'] )->name('showtasks');
+    Route::put('/update/{user}', [UserManagementController::class, 'update'] )->name('edituser');
+    Route::delete('/delete/{user}', [UserManagementController::class, 'delete'] )->name('delete');
```

Comment:
- Put this entire route list inside the  Route::middleware(['auth:admin'])->group(function(){}) so only login admin users can access them.

- Try grouping the routes according to its resource:
Example for tasks routes:
  Route::prefix("/tasks")->as('tasks.')->group(function(){
        Route::get('/', [TaskController::class, 'showAll'])->name('index');
        Route::get('/{user}', [UserManagementController::class, 'showTasks'] )->name('showtasks');
    });
   
-For update/delete methods,no need to put /update, /delete as its prefix. Do it like this:

    Route::put('/{user}', [UserManagementController::class, 'update'] )->name('edituser');
     Route::delete('/{user}', [UserManagementController::class, 'delete'] )->name('delete');